python/lsstdesc/pipe/task/metadetection.py

```diff
@@ -0,0 +1,112 @@
+# import lsst.daf.persistence as dafPersist  # Gen2 version
+from __future__ import annotations
+import typing
+
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+from lsst.pipe.base.struct import Struct
+import lsst.afw.table as afwTable
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+import lsst.utils
+
+
+class MetadetectConnections(pipeBase.PipelineTaskConnections,
+                            dimensions=("tract", "patch", "skymap"),
+                            # running all bands
+                            # not having instrument makes it possible to combine
+                            # calexp
+                           ):
+    calExpList = cT.Input(
+        doc="Input exposures to be resampled and optionally PSF-matched onto a SkyMap projection/patch",
+        name="calexp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    skyMap = pipeBase.connectionTypes.Input(
+        doc="Input definition of geometry/box and projection/wcs for coadded exposures",
+        name="skyMap",
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    # This is an output catalog that is not (yet) used in this skeleton
+    catalog = cT.Output(
+        doc=("Output catalog"),
+        name='metadetectObj',
+        storageClass="DataFrame",
+        dimensions=("tract", "patch", "skymap"),
+    )
+    coadd = cT.Output(
+        doc=("Coadded image"),
+        name="ngmixCoadd",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "visit", "instrument")
+    )
+    # Come back and apply jointcal/fgcm later to the calexp s.
+
+
+class MetadetectConfig(pipeBase.PipelineTaskConfig, pipelineConnections=MetadetectConnections):
+    """ Configuration parameters for the `MetadetectTask`.
+    """
+
+    pass
+
+
+class MetadetectTask(pipeBase.PipelineTask):
+    """ Metadetection.
+    """
+
+    ConfigClass = MetadetectConfig
+    _DefaultName = "metadetect"
+
+    # @pipeBase.timeMethod
+    def run(self, calExpList: typing.List[lsst.afw.image.ExposureF], skyInfo: pipeBase.Struct) -> pipeBase.Struct:
```

Comment:
How can we split this list by band, so we can coadd each band separately?

If I do `calexp.get()` I can see the filter but not before I load it

Reply:
At what stages will the different bands come together? If we're coadding each band separately (and not making a measurement on a multiband coadd), we could invoke a task with only tracts and patches from a single band. Let me check what API exists to see the calexp info without loading it.

Reply:
OK, so calexp.dataId['band'] will give the abstract band like `z` for the example, and calexp.dataId['physical_filter'] will output `HSC-Z`.